bloggitt/core/views.py

```diff
@@ -48,9 +48,47 @@ def postlist(request):
     post_list=Post.objects.all().order_by('-created_on')
     return render(request,'index.html',{'post_list':post_list})
 
+
 def postdetail(request, slug):
+    if not request.user.is_authenticated:
+        return redirect('login')
+        
     post = Post.objects.get(slug=slug)
-    return render(request, 'detail.html', {'post': post})
+    Favourites,_ = FavouritePost.objects.get_or_create(user=request.user)
+    post_in_favorites = None
+    if post in Favourites.posts.all():
+        post_in_favorites = True
+    else:
+        post_in_favorites = False
+
+    return render(request, 'detail.html', {'post': post, 'post_in_favorites': post_in_favorites})
+
+
+def Favorites(request, slug):
+    if not request.user.is_authenticated:
+        return redirect('login')
+
+    user = request.user
+    Favourites,_ = FavouritePost.objects.get_or_create(user=user)
+
+    post = Post.objects.get(slug=slug)
+
+    if post not in Favourites.posts.all():
+        Favourites.posts.add(post)
+        Favourites.save()
+    else:
+        Favourites.posts.remove(post)
+        Favourites.save()
+    
+    return redirect('favourites')
```

Comment:
Whenever a user adds or removes a post from his favorites, he should stay on the same page and the button should change accordingly. Why are you redirecting him to favourites page? Please look into this.

Reply:
Ok, i will update that

Reply:
I have changed the redirect to `post_detail` so as to stay in same page.